great_expectations/checkpoint/checkpoint.py

```diff
@@ -640,11 +640,14 @@ def action_list(self) -> List[Dict]:
             return []
 
     @property
-    def validations(self) -> List[CheckpointValidationConfig]:
+    def validations(
+        self,
+    ) -> list[CheckpointValidationConfig] | list[dict]:  # always a list[dict]
```

Comment:
If this is always `list[dict]`, why the union? Could we maybe add a line adding the context we discussed (if you deem in applicable)?

Reply:
Because our Checkpoint code is unjustifiably complex, unfortunately `Union` is needed (I filed a comprehensive JIRA ticket, which poses more questions than offering solutions; hence, it should be discussed).  Thanks.